db/user.go

```diff
@@ -221,3 +222,74 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to map a User.Role to all User.Email's for that course.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input.
+// The function returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// The function returns an error if a course cannot be opened and the input contains roles or a *.
+// The function will issue a warning if it's given a valid course and an invalid role.
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    emailSet := map[string]any{};
+    roleSet := map[string]any{};
+
+    // Iterate over every given string, checking for emails, roles, and *, which denotes all roles.
+    for _, email := range emails {
```

Comment:
Go ahead and trim/strip `email` as the first thing inside this loop (even before the empty check).
Since emails and roles can't have whitespace, it's safe.